api/paidAction/itemUpdate.js

```diff
@@ -0,0 +1,155 @@
+import { USER_ID } from '@/lib/constants'
+import { imageFeesInfo } from '../resolvers/image'
+import { getItemMentions, getMentions, performBotBehavior } from './lib/item'
+import { notifyItemMention, notifyMention } from '@/lib/webPush'
+import { satsToMsats } from '@/lib/format'
+
+export const anonable = false
+export const supportsPessimism = true
+export const supportsOptimism = false
+
+export async function getCost ({ id, boost = 0, uploadIds }, { me, models }) {
+  // the only reason updating items costs anything is when it has new uploads
+  // or more boost
+  const old = await models.item.findUnique({ where: { id: parseInt(id) } })
+  const { totalFeesMsats } = await imageFeesInfo(uploadIds, { models, me })
+  return BigInt(totalFeesMsats) + satsToMsats(boost - (old.boost || 0))
+}
+
+export async function perform (args, context) {
+  const { id, boost = 0, uploadIds = [], options: pollOptions = [], forwardUsers: itemForwards = [], invoiceId, ...data } = args
+  const { tx, me, models } = context
+  const old = await tx.item.findUnique({
+    where: { id: parseInt(id) },
+    include: {
+      threadSubscriptions: true,
+      mentions: true,
+      itemForwards: true,
+      itemReferrers: true,
+      itemUploads: true
+    }
+  })
+
+  const boostMsats = satsToMsats(boost - (old.boost || 0))
+  const itemActs = []
+  if (boostMsats > 0) {
+    itemActs.push({
+      msats: boostMsats, act: 'BOOST', userId: me?.id || USER_ID.anon
+    })
+  }
+
+  // createMany is the set difference of the new - old
+  // deleteMany is the set difference of the old - new
+  // updateMany is the intersection of the old and new
+  const difference = (a = [], b = [], key = 'userId') => a.filter(x => !b.find(y => y[key] === x[key]))
+  const intersectionMerge = (a = [], b = [], key) => a.filter(x => b.find(y => y.userId === x.userId))
+    .map(x => ({ [key]: x[key], ...b.find(y => y.userId === x.userId) }))
```

Comment:
We can't control the order of nested create, updates, and deletes in prisma. Besides, this is more accurate.